types/mempool/sender_nonce.go

```diff
@@ -50,8 +64,13 @@ func NewSenderNonceMempool() Mempool {
 // NewSenderNonceMempoolWithSeed creates a new mempool that prioritizes transactions by nonce, the lowest first and sets the random seed.
 func NewSenderNonceMempoolWithSeed(seed int64) Mempool {
 	senderMap := make(map[string]*skiplist.SkipList)
+	existingTx := make(map[txKey]bool)
 	snp := &senderNonceMempool{
-		senders: senderMap,
+		senders:    senderMap,
+		txCount:    0,
+		unbounded:  true,
+		maxTx:      0,
+		existingTx: existingTx,
```

Comment:
could you explain the use case of this?

Reply:
So when a tx for a sender has the same nonce, the current behavior is to overwrite the tx existing in the mempool. Therefore there are some cases where even though a new tx was inserted the count should not increase.

Reply:
makes sense, thanks for the explanation